packages/sdk-client-v3/src/middleware/create-concurrent-modification-middleware.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  Next,
+  Middleware,
+  MiddlewareRequest,
+  MiddlewareResponse,
+} from '../types/types'
+
+export default function createConcurrentModificationMiddleware(): Middleware {
+  return (next: Next) => {
+    return async (request: MiddlewareRequest): Promise<MiddlewareResponse> => {
+      const response = await next(request)
+      if (response.statusCode == 409) {
+        /**
+         * extract the currentVersion
+         * from the error body and update
+         * request with the currentVersion
+         */
+        const version = response.error?.body?.errors?.[0]?.currentVersion
+
+        // update the resource version here
+        request.body =
+          typeof request.body == 'string'
+            ? { ...JSON.parse(request.body), version }
+            : { ...request.body, version }
+
+        return next(request)
```

Comment:
shouldn't this be awaited?

```suggestion
        return await next(request)
```

Reply:
This await is redundant because in javascript return also serves as an implicit `await`